# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/exim/hostcc.patch.cross
# Copyright (C) 2022 - 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- exim-4.100.1.orig/OS/Makefile-Base	2026-10-17 21:50:43.646386310 +0200
+++ exim-4.100.1/OS/Makefile-Base	2026-10-17 21:50:43.655425857 +0200
@@ -127,114 +127,114 @@ OBJ_MACRO = macro_predef.o \
 $(OBJ_MACRO):	$(MACRO_HSRC)
 
 macro_predef.o :	macro_predef.c
-	@echo "$(CC) -DMACRO_PREDEF macro_predef.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ macro_predef.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF macro_predef.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ macro_predef.c
 macro-globals.o :	globals.c
-	@echo "$(CC) -DMACRO_PREDEF globals.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ globals.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF globals.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ globals.c
 macro-readconf.o :	readconf.c
-	@echo "$(CC) -DMACRO_PREDEF readconf.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ readconf.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF readconf.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ readconf.c
 macro-expand.o :	expand.c
-	@echo "$(CC) -DMACRO_PREDEF expand.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ expand.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF expand.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ expand.c
 macro-route.o :		route.c
-	@echo "$(CC) -DMACRO_PREDEF route.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ route.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF route.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ route.c
 macro-transport.o:	transport.c
-	@echo "$(CC) -DMACRO_PREDEF transport.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ transport.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF transport.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ transport.c
 macro-drtables.o :	drtables.c
-	@echo "$(CC) -DMACRO_PREDEF drtables.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ drtables.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF drtables.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ drtables.c
 macro-acl.o:	acl.c
-	@echo "$(CC) -DMACRO_PREDEF acl.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ acl.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF acl.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ acl.c
 macro-tls.o:	tls.c tls-gnu.c tls-openssl.c
-	@echo "$(CC) -DMACRO_PREDEF tls.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ tls.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF tls.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ tls.c
 macro-appendfile.o :	transports/appendfile.c
-	@echo "$(CC) -DMACRO_PREDEF transports/appendfile.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ transports/appendfile.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF transports/appendfile.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ transports/appendfile.c
 macro-autoreply.o :	transports/autoreply.c
-	@echo "$(CC) -DMACRO_PREDEF transports/autoreply.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ transports/autoreply.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF transports/autoreply.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ transports/autoreply.c
 macro-lmtp.o:		transports/lmtp.c
-	@echo "$(CC) -DMACRO_PREDEF transports/lmtp.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ transports/lmtp.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF transports/lmtp.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ transports/lmtp.c
 macro-pipe.o :		transports/pipe.c
-	@echo "$(CC) -DMACRO_PREDEF transports/pipe.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ transports/pipe.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF transports/pipe.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ transports/pipe.c
 macro-queuefile.o :	transports/queuefile.c
-	@echo "$(CC) -DMACRO_PREDEF transports/queuefile.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ transports/queuefile.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF transports/queuefile.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ transports/queuefile.c
 macro-smtp.o :		transports/smtp.c
-	@echo "$(CC) -DMACRO_PREDEF transports/smtp.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ transports/smtp.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF transports/smtp.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ transports/smtp.c
 macro-accept.o :	routers/accept.c
-	@echo "$(CC) -DMACRO_PREDEF routers/accept.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ routers/accept.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF routers/accept.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ routers/accept.c
 macro-dnslookup.o :	routers/dnslookup.c
-	@echo "$(CC) -DMACRO_PREDEF routers/dnslookup.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ routers/dnslookup.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF routers/dnslookup.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ routers/dnslookup.c
 macro-ipliteral.o :	routers/ipliteral.c
-	@echo "$(CC) -DMACRO_PREDEF routers/ipliteral.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ routers/ipliteral.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF routers/ipliteral.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ routers/ipliteral.c
 macro-iplookup.o :	routers/iplookup.c
-	@echo "$(CC) -DMACRO_PREDEF routers/iplookup.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ routers/iplookup.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF routers/iplookup.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ routers/iplookup.c
 macro-manualroute.o :	routers/manualroute.c
-	@echo "$(CC) -DMACRO_PREDEF routers/manualroute.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ routers/manualroute.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF routers/manualroute.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ routers/manualroute.c
 macro-queryprogram.o :	routers/queryprogram.c
-	@echo "$(CC) -DMACRO_PREDEF routers/queryprogram.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ routers/queryprogram.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF routers/queryprogram.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ routers/queryprogram.c
 macro-redirect.o :	routers/redirect.c
-	@echo "$(CC) -DMACRO_PREDEF routers/redirect.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ routers/redirect.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF routers/redirect.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ routers/redirect.c
 macro-auth-spa.o :	auths/auth-spa.c
-	@echo "$(CC) -DMACRO_PREDEF auths/auth-spa.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ auths/auth-spa.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF auths/auth-spa.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ auths/auth-spa.c
 macro-cram_md5.o :	auths/cram_md5.c
-	@echo "$(CC) -DMACRO_PREDEF auths/cram_md5.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ auths/cram_md5.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF auths/cram_md5.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ auths/cram_md5.c
 macro-cyrus_sasl.o :	auths/cyrus_sasl.c
-	@echo "$(CC) -DMACRO_PREDEF auths/cyrus_sasl.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ auths/cyrus_sasl.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF auths/cyrus_sasl.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ auths/cyrus_sasl.c
 macro-dovecot.o:	auths/dovecot.c
-	@echo "$(CC) -DMACRO_PREDEF auths/dovecot.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ auths/dovecot.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF auths/dovecot.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ auths/dovecot.c
 macro-external.o:	auths/external.c
-	@echo "$(CC) -DMACRO_PREDEF auths/external.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ auths/external.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF auths/external.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ auths/external.c
 macro-gsasl.o :	auths/gsasl.c
-	@echo "$(CC) -DMACRO_PREDEF auths/gsasl.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ auths/gsasl.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF auths/gsasl.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ auths/gsasl.c
 macro-heimdal_gssapi.o:	auths/heimdal_gssapi.c
-	@echo "$(CC) -DMACRO_PREDEF auths/heimdal_gssapi.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ auths/heimdal_gssapi.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF auths/heimdal_gssapi.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ auths/heimdal_gssapi.c
 macro-plaintext.o :	auths/plaintext.c
-	@echo "$(CC) -DMACRO_PREDEF auths/plaintext.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ auths/plaintext.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF auths/plaintext.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ auths/plaintext.c
 macro-spa.o :		auths/spa.c
-	@echo "$(CC) -DMACRO_PREDEF auths/spa.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ auths/spa.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF auths/spa.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ auths/spa.c
 macro-authtls.o:	auths/tls.c
-	@echo "$(CC) -DMACRO_PREDEF auths/tls.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ auths/tls.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF auths/tls.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ auths/tls.c
 macro-dkim.o:		miscmods/dkim.c
-	@echo "$(CC) -DMACRO_PREDEF miscmods/dkim.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ miscmods/dkim.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF miscmods/dkim.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ miscmods/dkim.c
 macro-malware.o:	malware.c
-	@echo "$(CC) -DMACRO_PREDEF malware.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ malware.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF malware.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ malware.c
 macro-signing.o:	miscmods/signing.c
-	@echo "$(CC) -DMACRO_PREDEF miscmods/signing.c"
-	$(FE)$(CC) -c $(CFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ miscmods/signing.c
+	@echo "$(HOSTCC) -DMACRO_PREDEF miscmods/signing.c"
+	$(FE)$(HOSTCC) -c $(HOSTCFLAGS) -DMACRO_PREDEF $(INCLUDE) -o $@ miscmods/signing.c
 
 macro_predef: $(OBJ_MACRO)
-	@echo "$(LNCC) -o $@"
-	$(FE)$(LNCC) -o $@ $(LFLAGS) $(OBJ_MACRO)
+	@echo "$(HOSTCC) -o $@"
+	$(FE)$(HOSTCC) -o $@ $(OBJ_MACRO)
 
 macro.c: macro_predef
 	./macro_predef > tmpfile
@@ -257,8 +257,8 @@ utils: $(EXIM_MONITOR) exicyclog exinext
 
 # Targets for special-purpose configuration header builders
 buildconfig: buildconfig.c
-	@echo "$(CC) buildconfig.c"
-	$(FE)$(CC) $(CFLAGS) $(INCLUDE) -o buildconfig buildconfig.c $(LIBS)
+	@echo "$(HOSTCC) buildconfig.c"
+	$(FE)$(HOSTCC) $(HOSTCFLAGS) $(INCLUDE) -o buildconfig buildconfig.c $(LIBS)
 
 
 UTIL_DEPS = $(CONFIG_DEPS) ../src/utils/msgid.frag
